c2corg_ui/static/js/yeti/yeti.js

```diff
@@ -0,0 +1,447 @@
+  this.scope_.$watchGroup([
+    'method',
+    'bra.haut',
+    'bra.bas',
+    'bra.altiseuil',
+    'bra.isDifferent',
+    'mapZoomOK'
+  ], (newValues, oldValues, scope) => {
+    // verif form
+    if (!this.scope_['bra']['haut']) {
+      this.formOK = false;
+      this.formOKError_ = 'bra';
+    } else if (!this.scope_['method']) {
+      this.formOK = false;
+      this.formOKError_ = 'methode';
+    } else if (
+      this.scope_['bra']['haut'] === 4 &&
+      this.scope_['method'] === 'mrd') {
+      this.formOK = false;
+      this.formOKError_ = 'methode_bra';
+    } else if (
+      this.scope_['bra']['bas'] &&
+      (this.scope_['bra']['haut'] != this.scope_['bra']['bas']) &&
+      !this.scope_['bra']['altiseuil']) {
+      this.formOK = false;
+      this.formOKError_ = 'altitude';
+    } else if (!this.scope_['mapZoomOK']) {
+      this.formOK = false;
+      this.formOKError_ = 'zoom';
+    } else {
+      this.formOK = true;
+    }
+    // set actual error
+    if (this.formOK) {
+      this.actualError = this.errors_['ok'];
+    } else {
+      this.actualError = this.errors_[this.formOKError_]['simple'];
+      if (this.formOKError_ === 'zoom') {
+        this.actualError += ' (actuel: ' + this.map_.getView().getZoom() + ')';
+      }
+    }
+    // also
+    // verif if bra = 4, method MRD forbidden
+    if (this.scope_['bra']['haut'] == 4 && this.scope_['method'] === 'mrd') {
+      delete this.scope_['method'];
+    }
+    // verif bra.isDifferent: empty inputs
+    if (!this.scope_['bra']['isDifferent']) {
+      delete this.scope_['bra']['bas'];
+      delete this.scope_['bra']['altiseuil'];
+    }
+  });
+};
+
+/**
+ * Set potentielDanger
+ * @private
+ */
+app.YetiController.prototype.setDanger_ = function() {
+  this.scope_['potentielDanger'] = this.DANGER.min;
+  this.scope_['potentielDangerMin'] = this.DANGER.min;
+  this.scope_['potentielDangerMax'] = this.DANGER.max;
+  this.scope_['potentielDangerStyle'] = {
+    width: 'calc(100% - (100% / ' + this.DANGER.max + ' / 2 - 10px) * 2)',
+    marginLeft: 'calc(100% / ' + this.DANGER.max + ' / 2 - 10px)'
+  };
+  this.scope_['potentielDangerLabel'] = [];
+  for (let i = 1; i <= this.scope_['potentielDangerMax']; i++) {
+    const data = {};
+    data['nb'] = i;
+    if (i === 2 || i === 4 || i === 8 || i === 16) {
+      data['val'] = i;
+    }
+    this.scope_['potentielDangerLabel'].push(data);
+  }
+  this.scope_.$watch('bra.haut', (newValue, oldValue) => {
+    if (newValue) {
+      const min = this.DANGER.bra[newValue - 1].min;
+      const max = this.DANGER.bra[newValue - 1].max;
+      const val = this.DANGER.bra[newValue - 1].val;
+
+      this.scope_['potentielDangerMin'] = min;
+      this.scope_['potentielDangerMax'] = max;
+
+      // compute width / margin-left
+      this.scope_['potentielDangerStyle'] = {
+        width: 'calc(' + (((max - min) + 1) * 100 / this.DANGER.max) + '% - (100% / ' + this.DANGER.max + ' / 2 - 10px) * 2)',
```

Comment:
The computation is the same when initializing stuff, except that `min` and `max` are no set. Maybe it could be factorized?